sdk/eventhub/event-hubs/src/diagnostics/tracing.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { createSpanFunction, SpanOptions } from "@azure/core-tracing";
+import { Span, SpanKind } from "@opentelemetry/api";
+import { EventHubConnectionConfig } from "../eventhubConnectionConfig";
+import { OperationOptions } from "../util/operationOptions";
+
+const _createSpan = createSpanFunction({
+  namespace: "Microsoft.EventHub",
+  packagePrefix: "Azure.EventHubs"
+});
+
+/**
+ * Creates an EventHubs specific span, with peer.address and message_bus.destination filled out.
+ * @internal
+ */
+export function createEventHubSpan(
+  operationName: string,
+  operationOptions: OperationOptions | undefined,
+  connectionConfig: Pick<EventHubConnectionConfig, "entityPath" | "host">,
+  additionalSpanOptions?: SpanOptions
+): { span: Span; updatedOptions: OperationOptions } {
+  const { span, updatedOptions } = _createSpan(operationName, {
```

Comment:
Is there some mutation happening that requires us to spread all of the tracing and span options into new objects?

Reply:
There are two "common" ones with this function - either someone wants to set the .kind to something other than INTERNAL or adding in links.